indico/modules/events/papers/lists.py

```diff
@@ -0,0 +1,142 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2017 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+from operator import attrgetter
+
+from flask import request
+from sqlalchemy.orm import subqueryload
+
+from indico.core.db import db
+from indico.modules.events.contributions import Contribution
+from indico.modules.events.papers.models.revisions import PaperRevisionState, PaperRevision
+from indico.modules.events.util import ListGeneratorBase
+from indico.util.i18n import _
+from indico.web.flask.templating import get_template_module
+
+
+class PaperAssignmentListGenerator(ListGeneratorBase):
+    """Listing and filtering actions in a paper assignment list."""
+
+    endpoint = '.assignment'
+    list_link_type = 'paper_asssignment_management'
+
+    def __init__(self, event):
+        super(PaperAssignmentListGenerator, self).__init__(event)
+        self.default_list_config = {
+            'items': ('state',),
+            'filters': {'items': {}}
+        }
+
+        state_not_submitted = {None: _('Not yet submitted')}
+        track_empty = {None: _('No track')}
+        session_empty = {None: _('No session')}
+        type_empty = {None: _('No type')}
+        state_choices = {state.value: state.title for state in PaperRevisionState}
+        track_choices = OrderedDict((unicode(t.id), t.title) for t in sorted(self.event.tracks,
```

Comment:
most likely it goes through json encoding at some point and JSON only supports string keys